pkgs/top-level/all-packages.nix

```diff
@@ -25148,6 +25148,7 @@ with pkgs;
 
   mastodon = callPackage ../servers/mastodon {
     nodejs-slim = nodejs-slim_20;
+    python3 = python310;
```

Comment:
Why 3.10? I remember pinning specific versions is good, but why not use the current 3.12 then?

Reply:
Error build with Python 3.12:
```
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info it worked if it ends with ok
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info using node-gyp@10.0.1
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info using node@20.15.1 | linux | x64
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info find Python using Python version 3.12.4 found at "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python3"
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn /nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python3
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args [
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '/build/source/node_modules/node-gyp/gyp/gyp_main.py',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args 'binding.gyp',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-f',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args 'make',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-I',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '/build/source/node_modules/utf-8-validate/build/config.gypi',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-I',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '/build/source/node_modules/node-gyp/addon.gypi',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-I',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '/nix/store/ypcn728wnqblyapkdix3hywdy574xd22-nodejs-slim-20.15.1/include/node/common.gypi',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dlibrary=shared_library',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dvisibility=default',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dnode_root_dir=/nix/store/ypcn728wnqblyapkdix3hywdy574xd22-nodejs-slim-20.15.1',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dnode_gyp_dir=/build/source/node_modules/node-gyp',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dnode_lib_file=/nix/store/ypcn728wnqblyapkdix3hywdy574xd22-nodejs-slim-20.15.1/$(Configuration)/node.lib',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dmodule_root_dir=/build/source/node_modules/utf-8-validate',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dnode_engine=v8',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '--depth=.',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '--no-parallel',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '--generator-output',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args 'build',
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Goutput_dir=.'
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args ]
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR <string>:17: SyntaxWarning: invalid escape sequence '\.'
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn make
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT make: Entering directory '/build/source/node_modules/utf-8-validate/build'
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT   CXX(target) Release/obj.target/validation/src/validation.o
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT   CXX(target) Release/obj.target/validation/deps/is_utf8/src/is_utf8.o
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT   SOLINK_MODULE(target) Release/obj.target/validation.node
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT   COPY Release/validation.node
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT make: Leaving directory '/build/source/node_modules/utf-8-validate/build'
mastodon-modules> ➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info ok

```